src/WhereConditions/WhereHasConditionsDirective.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Nuwave\Lighthouse\WhereConditions;
+
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\FieldDefinitionNode;
+use GraphQL\Language\AST\InputValueDefinitionNode;
+use GraphQL\Language\AST\ObjectTypeDefinitionNode;
+use Illuminate\Support\Str;
+use Nuwave\Lighthouse\Schema\AST\Codegen;
+use Nuwave\Lighthouse\Schema\AST\DocumentAST;
+use Nuwave\Lighthouse\Schema\AST\PartialParser;
+use Nuwave\Lighthouse\Schema\Directives\BaseDirective;
+use Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective;
+use Nuwave\Lighthouse\Support\Contracts\ArgManipulator;
+use Nuwave\Lighthouse\Support\Contracts\DefinedDirective;
+
+class WhereHasConditionsDirective extends BaseDirective implements ArgBuilderDirective, ArgManipulator, DefinedDirective
+{
+    /**
+     * @var \Nuwave\Lighthouse\WhereConditions\Operator
+     */
+    protected $operator;
+
+    /**
+     * WhereHasConditionssDirective constructor.
+     *
+     * @param  \Nuwave\Lighthouse\WhereConditions\Operator  $operator
+     * @return void
+     */
+    public function __construct(Operator $operator)
+    {
+        $this->operator = $operator;
+    }
+
+    public static function definition(): string
+    {
+        return /* @lang GraphQL */ <<<'SDL'
+"""
+Add a dynamically client-controlled relationship WHERE condition to a fields query.
+The argument it is defined on may have any name but **must** be
+of the input type `WhereHasConditions`.
+"""
+directive @whereHasConditions(
+    """
+    The Eloquent relationship that the conditions will be applied to.
+    This argument can be ommited if the field name and the relationship name are the same.
```

Comment:
I think we should actually go for the `hasXyz` style naming convention. Is is a bit more work in terms of docs and implementation, but i think it encourages much better naming by default.

Reply:
Yeah, I agree that the `hasXyz` naming convention is more descriptive to the client than the `xyz` naming convention. I'll add this.

Reply:
Done!